capstone/scripts/convert_s3.py

```diff
@@ -0,0 +1,403 @@
+            )
+
+
+# Case-specific helper functions
+
+
+def case_content_exists_in_s3(bucket, key, file):
+    """
+    Check whether file with the same content already exists in S3 using
+    ChecksumSHA256 hash comparison.
+    """
+    try:
+        # Calculate the SHA256 hash of the file content
+        file_data = file.read()
+        hash_object = hashlib.sha256(file_data)
+        sha256_hash = base64.b64encode(hash_object.digest()).decode()
+
+        # Get the object's metadata
+        response = s3_client.get_object_attributes(
+            Bucket=bucket, Key=key, ObjectAttributes=["Checksum"]
+        )
+
+        existing_hash = response.get("Checksum", {}).get("ChecksumSHA256")
+
+        # Compare the hashes
+        return existing_hash == sha256_hash
+
+    except ClientError as err:
+        if err.response["Error"]["Code"] == "404":
+            return False
+        else:
+            raise Exception(f"Cannot check file {bucket}/{key}: %s" % err)
+
+
+def build_unique_case_name(first_page, bucket, case_prefix, file):
+    """
+    Create unique case name based on first page
+    """
+    suffix = 1
+    first_page = int(first_page)
+    case_name = f"{first_page:04d}-{suffix:02d}"
+
+    while case_name_exists(case_name, bucket, case_prefix):
+        if case_content_exists_in_s3(bucket, f"{case_prefix}/{case_name}.json", file):
+            print(f"Skipping {case_prefix}/{case_name}. Duplicate content.")
+            return None
+        suffix += 1
+        case_name = f"{first_page:04d}-{suffix:02d}"
+
+    return case_name
+
+
+def case_name_exists(case_name, bucket, case_prefix):
+    """
+    Find out whether the case name exists in S3
+    """
+    response = s3_client.list_objects_v2(
+        Bucket=bucket, Prefix=f"{case_prefix}/{case_name}"
+    )
+    objects = response.get("Contents", [])
+    return bool(objects)
+
+
+def case_content_exists(bucket, key):
+    try:
+        s3_client.head_object(Bucket=bucket, Key=key)
+        return True
+    except ClientError as err:
+        if err.response["Error"]["Code"] == "404":
+            return False
+        else:
+            raise Exception(f"Cannot check file {bucket}/{key}: %s" % err)
+
+
+# General helper functions
+
+
+def hash_and_upload(file, bucket, key, content_type):
+    """
+    Hash created file and upload to S3
+    """
+    with open(file.name, "rb") as file:
+        # read the file's contents
+        file_data = file.read()
+    # Calculate the SHA256 hash of the file data
+    hash_object = hashlib.sha256(file_data)
+    sha256_hash = base64.b64encode(hash_object.digest()).decode()
+    # upload file to S3
+    try:
+        s3_client.put_object(
+            Body=file_data,
+            Bucket=bucket,
+            Key=key,
+            ContentType=content_type,
+            ChecksumSHA256=sha256_hash,
+        )
+    except ClientError as err:
+        raise Exception(f"Error uploading {key}: %s" % err)
+
+
+def get_bucket_name(redacted):
```

Comment:
I think type annotations could give a clarity boost to this script. 
For example: 
```python
def get_bucket_name(redacted: bool) -> str:
```

Tells me a lot more about what this function does at a glance.